src/components/v5/common/ActionSidebar/ActionSidebar.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { FC, PropsWithChildren, useRef, useState } from 'react';
+import clsx from 'clsx';
+import { useIntl } from 'react-intl';
+
+import { useOnClickOutside } from 'usehooks-ts';
+import styles from './ActionSidebar.module.css';
+import Icon from '~shared/Icon';
+import { useMobile } from '~hooks';
+import Button, { TextButton } from '~v5/shared/Button';
+import ActionSidebarRow from '../ActionSidebarRow';
+import { useActionSidebarContext } from '~context/ActionSidebarContext';
+
+const ActionSidebar: FC<PropsWithChildren> = ({ children }) => {
+  const ref = useRef(null);
+  const [isSidebarFullscreen, setIsSidebarFullscreen] = useState(false);
```

Comment:
You could create a useToggle hook for easier control over boolean state:
```
useToggle(defaultValue?: boolean): {
  value: boolean;
  toggle(): void;
  toggleOn(): void;
  toggleOff(): void;
}
```
You could also reuse it in useActionSidebarContext.